smart_contracts/contracts/test/list-authorization-keys/src/main.rs

```diff
@@ -0,0 +1,34 @@
+#![no_std]
+#![no_main]
+
+extern crate alloc;
+
+use alloc::collections::BTreeSet;
+
+use casper_contract::contract_api::runtime;
+use casper_types::{account::AccountHash, ApiError};
+
+const ARG_EXPECTED_AUTHORIZATION_KEYS: &str = "expected_authorized_keys";
+
+#[repr(u16)]
+enum UserError {
+    AssertionFail = 0,
+}
+
+impl From<UserError> for ApiError {
+    fn from(error: UserError) -> ApiError {
+        ApiError::User(error as u16)
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn call() {
```

Comment:
i think we should also add a more complex test contract that sets up the userland use case and proves it works as expected

such a contract would define two roles...lets say role A and role B, with an associated account hash assigned to each, and have two entrypoints...A and B. 

entrypoint A passes if account hash associated to role A signed the deploy calling it else fails

entrypoint B passes if account hash associated to role B signed the deploy calling it else fails

Reply:
and by fail I mean reverts with a custom user error, in case that wasn't obvious

Reply:
Addressed in https://github.com/mpapierski/casper-node/blob/4001974acc48f423064671c1feab4a24583c63d0/smart_contracts/contracts_as/test/multisig-authorization/src/main.rs#L1-L104